lib/executor/vu_handle.go

```diff
@@ -101,15 +102,11 @@ func (vh *vuHandle) hardStop() {
 
 //TODO: simplify this somehow - I feel like there should be a better way to
 //implement this logic... maybe with sync.Cond?
-func (vh *vuHandle) runLoopsIfPossible(runIter func(context.Context, lib.VU)) {
+func (vh *vuHandle) runLoopsIfPossible(runIter func(context.Context, lib.ActiveVU)) {
 	executorDone := vh.parentCtx.Done()
 
-	var vu lib.VU
-	defer func() {
-		if vu != nil {
-			vh.returnVU(vu)
-		}
-	}()
```

Comment:
My testing showed that you are getting variable looping vus stuck at waiting for all vus to be returned. And I would wager it is this change, and the ones below, that are actually at the root of it

Reply:
below you have a couple of returns at which you don't return the VUs ... I am not particularly certain if VLV executor needs to wait for them, but still they should be returned :D , and I find it waiting to probably be better then ... not waiting :D

Reply:
So I think this is actually correct as is, but other places needed fixing.

The idea of the VU deactivation is that the VU will be returned to the global pool when the `DeactivateCallback` is called. So that logic shouldn't be in `vuHandle` anymore.

But there were other issues that resulted in executors waiting for VUs to be returned:

- `MiniRunner.Activate` didn't handle deactivation at all, so VUs were never returned in tests that used it.
- Executors need to wait for all deactivation goroutines to finish, so that the callback is actually called.

Both issues were fixed in e1fcc20a, but based on the current CI results, it's clearly leaking somewhere (though it passed on AppVeyor :laughing:). Let me know if it's a step in the right direction, though.